src/main/kotlin/snyk/common/CustomEndpoints.kt

```diff
@@ -1,14 +1,16 @@
 package snyk.common
 
 import io.snyk.plugin.pluginSettings
-import io.snyk.plugin.prefixIfNot
 import io.snyk.plugin.suffixIfNot
 import java.net.URI
 import java.net.URISyntaxException
 
 fun toSnykCodeApiUrl(endpointUrl: String?): String {
     val endpoint = resolveCustomEndpoint(endpointUrl)
     val uri = URI(endpoint)
+    if (uri.isLocalCodeEngine()) {
```

Comment:
As said before, this should be moved into the when in line 16

Reply:
The only reason I kept it here is because we have no control on how SCLE URLs will look like, its 100% up to the customers. SCLE URLs does not follow our URL conventions.

The return at  line26/28 removes and replaces some text from the URL, this could potentially break a SCLE URL matching any of the rules in the suffix/replacement

Reply:
Why not:
```kotlin
fun toSnykCodeApiUrl(endpointUrl: String?): String {
    val endpoint = resolveCustomEndpoint(endpointUrl)
    val uri = URI(endpoint)

    val codeSubdomain = "deeproxy"
    val snykCodeApiUrl = when {
        uri.isLocalCodeEngine() ->
            return pluginSettings().localCodeEngineUrl!!
        
        uri.isDeeproxy() ->
            endpoint

        uri.isDev() ->
            endpoint.replace("https://dev.", "https://$codeSubdomain.dev.")

        uri.isSnykTenant() ->
            endpoint.replace("https://", "https://$codeSubdomain.")

        else -> "https://$codeSubdomain.snyk.io/"
    }
    return snykCodeApiUrl.removeSuffix("api").replace("app.", "")
}
```

Reply:
Done